tests/test_forms.py

```diff
@@ -33,3 +37,330 @@ class Meta:
         article.refresh_from_db()
         self.assertEqual(article.title_nl, "Nederlandse taal")
         self.assertEqual(article.title_en, "English language")
+
+
+class Form(TranslationModelForm):
+    """Challenge form."""
+
+    start_date = forms.DateField(label="start date")
+    end_date = forms.DateField(label="end date")
+
+    class Meta:
+        model = Challenge
+        fields = ["title", "start_date", "default_language", "header"]
+        required = {"title": True, "header": False}
+
+
+class ExcludeForm(TranslationModelForm):
+    """Challenge form."""
+
+    start_date = forms.DateField(label="start date")
+    end_date = forms.DateField(label="end date")
+
+    field_order = ["start_date", "title", "end_date"]  # test the custom field_order setting
+
+    class Meta:
+        model = Challenge
+        exclude = ["header", "default_language"]
+        languages = ["browser", "de", "fallback"]
+        fallback_language = "fr"
+        widgets = {"title": forms.widgets.Textarea}
+
+
+class TranslationFormTestCase(TestCase):
+    def test_languages_errors(self):
+        """Test the error messages for incorrect languages options."""
+
+        with self.assertRaisesMessage(ValueError, "languages: values should be strings"):
+            Form(languages=[0, "fallback"])
+
+        with self.assertRaisesMessage(ValueError, "languages: value es_it is not permitted"):
+            Form(languages=["es_it", "fallback"])
+
+        with self.assertRaisesMessage(ValueError, "languages: value e is not permitted"):
+            Form(languages=["e", "fallback"])
+
+        with self.assertRaisesMessage(
+            ValueError, "languages: xx is not an available language in the system"
+        ):
+            Form(languages=["xx", "fallback"])
```

Comment:
After the code producing this error is simplified, this could be simplified too:

```suggestion
        for language in [0, "es_it", "e", "xx"]:
            with self.assertRaisesMessage(ValueError, f"languages: {language} is not permitted"):
                Form(languages=[language])
```